src/adapters/scene_direction.js

```diff
@@ -53,6 +53,36 @@ export default class SceneDirection {
     listen('unhighlight_step', step => {
       this.unhighlightStep(step);
     });
+
+    listen('set_origin', poi => {
+      this.setOrigin(poi);
+    });
+
+    listen('set_destination', poi => {
+      this.setDestination(poi);
+    });
+  }
+
+  setOrigin = poi => {
+    const originMarker = createMarker(
+      poi.latLon,
+      'itinerary_marker_origin', { draggable: true }
+    )
+      .addTo(this.map)
+      .on('dragend', event => this.refreshDirection('origin', event.target.getLngLat()));
+    this.routeMarkers.push(originMarker);
+    fire('fit_map', poi);
```

Comment:
As you suggested, passing an extra argument to keep the current zoom in the `fitMap` method would be great.